src/scroll-timeline-base.js

```diff
@@ -527,6 +326,122 @@ function getScrollParent(node) {
+
+  const unresolved = null;
+  if (timeline.phase === 'inactive')
+    return unresolved;
+
+  if (!(timeline instanceof ViewTimeline))
+    return unresolved;
+
+  // Compute the offset of the top-left corner of subject relative to
+  // top-left corner of the container.
+  const container = timeline.source;
+  const target = timeline.subject;
+
+  let top = 0;
+  let left = 0;
+  let node = target;
+  const ancestor = container.offsetParent;
+  while (node && node != ancestor) {
+    left += node.offsetLeft;
+    top += node.offsetTop;
+    node = node.offsetParent;
+  }
+  left -= container.offsetLeft + container.clientLeft;
+  top -= container.offsetTop + container.clientTop;
+
+  // Determine the view and container size based on the scroll direction.
+  // The view position is the scroll position of the logical starting edge
+  // of the view.
+  const style = getComputedStyle(container);
+  const horizontalWritingMode = style.writingMode == 'horizontal-tb';
+  const rtl = style.direction == 'rtl' || style.writingMode == 'vertical-rl';
+  let viewSize = undefined;
+  let viewPos = undefined;
+  let containerSize = undefined;
+  const orientation = details.orientation;
+  if (orientation == 'horizontal' ||
+      (orientation == 'inline' && horizontalWritingMode) ||
+      (orientation == 'block' && !horizontalWritingMode)) {
+    viewSize = target.clientWidth;
+    viewPos = left;
+    if (rtl)
+      viewPos += container.scrollWidth - container.clientWidth;
+    containerSize = container.clientWidth;
+  } else {
+    // TODO: support sideways-lr
+    viewSize = target.clientHeight;
+    viewPos = top;
+    containerSize = container.clientHeight;
+  }
+
+  const scrollPos = directionAwareScrollOffset(container, orientation);
+  let startOffset = undefined;
+  let endOffset = undefined;
+
+  switch(phase) {
+    case 'cover':
+      // Range of scroll offsets where the subject element intersects the
+      // source's viewport.
+      startOffset = viewPos - containerSize;
+      endOffset = viewPos + viewSize;
+      break;
+
+    case 'contain':
+      // Range of scroll offsets where the subject element is fully inside of
+      // the container's viewport. If the subject's bounds exceed the size
+      // of the viewport in the scroll direction then the scroll range is
+      // empty.
+      startOffset = viewPos + viewSize - containerSize;
+      endOffset = viewPos;
+      break;
+
+    case 'enter':
+      // Range of scroll offsets where the subject element overlaps the
+      // logical-start edge of the viewport.
+      startOffset = viewPos - containerSize;
+      endOffset = viewPos + viewSize - containerSize;
+      break;
+
+    case 'exit':
+      // Range of scroll offsets where the subject element overlaps the
+      // logical-end edge of the viewport.
+      startOffset = viewPos;
+      endOffset = viewPos + viewSize;
+      break;
+  }
+
+  // TODO: Clamping of offsets is not specced. Update once ratified.
```

Comment:
Link to the issue - also the current leading proposal in issue is not to clamp so perhaps we should follow that.

Reply:
Added spec link.  Already landed code in Blink that clamps cover range.  I'd rather remain consistent with Blink implementation and update both sides if necessary once ratified on the spec.

Reply:
I don't think being consistent with Blink's implementation, when it doesn't match the spec is a goal. Using unclamped would allow my last comment to be a functional demo, whereas with clamped times it's not easy to emulate unclamped behavior.

Reply:
Done.